ReactAndroid/src/main/java/com/facebook/react/views/view/ReactViewGroup.java

```diff
@@ -455,6 +456,50 @@ private void updateSubviewClipStatus(View subview) {
     }
   }
 
+  @Override
+  public boolean getChildVisibleRect(View child, Rect r, android.graphics.Point offset) {
+    // This is based on the Android ViewGroup implementation, modified to clip child rects
+    // if overflow is set to ViewProps.HIDDEN. This effectively solves Issue #23870 which
+    // appears to have been introduced by FLAG_CLIP_CHILDREN being forced false
+    // regardless of whether clipping is desired.
+    final RectF rect = new RectF();
```

Comment:
@davidbiedenbach why give up on android's optimizations, such as:
https://android.googlesource.com/platform/frameworks/base/+/master/core/java/android/view/ViewGroup.java#6192
and: 
https://android.googlesource.com/platform/frameworks/base/+/master/core/java/android/view/ViewGroup.java#6195

Reply:
@d4vidi I tried to do it that way at first, but unfortunately this class doesn't have access to the superclass members required to use those optimizations.